extensions/podman/src/podman-cli.ts

```diff
@@ -56,87 +54,13 @@ export function getCustomBinaryPath(): string | undefined {
   return configuration.getConfiguration('podman').get('binary.path');
 }
 
-export interface ExecOptions {
-  context?: LifecycleContext;
-  logger?: Logger;
-  env?: NodeJS.ProcessEnv;
-}
-
-export function execPromise(
-  command: string,
-  args?: string[],
-  options?: ExecOptions,
-  token?: CancellationToken,
-): Promise<string> {
-  let env = Object.assign({}, process.env); // clone original env object
-
-  // In production mode, applications don't have access to the 'user' path like brew
-  if (isMac() || isWindows()) {
-    env.PATH = getInstallationPath();
-  } else if (env.FLATPAK_ID) {
-    // need to execute the command on the host
-    args = ['--host', command, ...args];
-    command = 'flatpak-spawn';
-  }
-
-  if (options?.env) {
-    env = Object.assign(env, options.env);
-  }
-  return new Promise((resolve, reject) => {
-    let stdOut = '';
-    let stdErr = '';
-    const process = spawn(command, args, { env });
-    // if the token is cancelled, kill the process and reject the promise
-    token?.onCancellationRequested(() => {
-      process.kill();
-      // reject the promise
-      options?.context?.log.error('Execution cancelled');
-      options?.logger?.error('Execution cancelled');
-      reject(new Error('Execution cancelled'));
-    });
-    process.on('error', error => {
-      let content = '';
-      if (stdOut && stdOut !== '') {
-        content += stdOut + '\n';
-      }
-      if (stdErr && stdErr !== '') {
-        content += stdErr + '\n';
-      }
-      options?.context?.log.error(content);
-      options?.logger?.error(content);
-      reject(new Error(content + error));
-    });
-    process.stdout.setEncoding('utf8');
-    process.stdout.on('data', data => {
-      stdOut += data;
-      options?.context?.log.log(data);
-      options?.logger?.log(data);
-    });
-    process.stderr.setEncoding('utf8');
-    process.stderr.on('data', data => {
-      stdErr += data;
-      options?.context?.log.warn(data);
-      options?.logger?.warn(data);
-    });
-
-    process.on('close', exitCode => {
-      if (exitCode !== 0) {
-        options?.context?.log.error(stdErr);
-        options?.logger?.error(stdErr);
-        reject(new Error(stdErr));
-      }
-      resolve(stdOut.trim());
-    });
-  });
-}
-
 export interface InstalledPodman {
   version: string;
 }
 
 export async function getPodmanInstallation(): Promise<InstalledPodman | undefined> {
   try {
-    const versionOut = await execPromise(getPodmanCli(), ['--version']);
+    const versionOut = (await process.exec(getPodmanCli(), ['--version'])).stdout;
```

Comment:
wondering if we don't need some small commands like getPodmanCliVersion() , getPodmanMachineList(), etc

Reply:
as far as this method is used ones, not sure, that we need a dedicate function for that, unless this call will be used somewhere else